docs/assertions.md

```diff
@@ -153,6 +153,27 @@ function test_failure() {
 ```
 :::
 
+## assert_line_count
+> `assert_line_count "count" "haystack"`
+
+Reports an error if `haystack` does not contain `count` lines.
+
+::: code-group
+```bash [Example]
+function test_success() {
+  local string="this is line one
+this is line two
+this is line three"
+
+  assert_line_count "3" "$string"
```

Comment:
Can we use literal numbers in the example instead of strings? I assume it would work, and it makes more sense. What do you think? (Same for line 172 here, and in the unit tests)
```suggestion
  assert_line_count 3 "$string"
```

Reply:
fixed, thanks

Reply:
btw: just noticed in other examples - e.g. `assert_less_than` - we use quotes arround numbers

Reply:
I guess for bash is the same (everything is a string...), and probably there is a mix between the old and new codebase. That said, we would like to keep a bit `(strict_types=1)` mindset in the project from now on. Feel free to submit a refactoring PR for this if you see any opportunity! :)